plugins/woocommerce/src/Caches/OrderAggregateCache.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+
+/**
+ * A class to cache aggregates from orders.
+ */
+class OrderAggregateCache {
+
+	/**
+	 * Option name.
+	 */
+	const OPTION_NAME = 'woocommerce_orders_aggregate_cache';
+
+	/**
+	 * The running totals for all order statuses.
+	 *
+	 * @var array|null
+	 */
+	private $totals = null;
+
+	/**
+	 * Whether or not the count has changed since loading.
+	 *
+	 * @var bool
+	 */
+	private $has_changed = false;
+
+	/**
+	 * Get the aggregate by identifier.
+	 *
+	 * @param string|string[] $status The order status to count.
+	 * @return int
+	 * @throws \Exception Exception thrown if status it not valid.
+	 */
+	public function get_count( $status ): int {
+		$status         = (array) $status;
+		$valid_statuses = $this->get_valid_statuses();
+
+		if ( ! empty( array_diff( $status, $valid_statuses ) ) ) {
+			throw new \Exception();
+		}
+
+		if ( ! $this->totals ) {
+			$cache        = get_option( self::OPTION_NAME );
+			$this->totals = $cache ? $cache : $this->count_orders_by_status();
+		}
+
+		return array_sum( array_intersect_key( $this->totals, array_flip( $status ) ) );
+	}
+
+	/**
+	 * Get valid statuses.
+	 *
+	 * @return string[]
+	 */
+	private function get_valid_statuses() {
+		return array_merge(
+			array_keys( wc_get_order_statuses() ),
+			array(
+				'trash',
```

Comment:
Implementation related: we should use constants from `\Automattic\WooCommerce\Enums\OrderStatus` here.

Reply:
Good call.  Updated.